tests/integration_tests/test_wandb.py

```diff
@@ -59,15 +87,49 @@ def test_log_api_call_count(wandb: mock.Mock) -> None:
     assert wandb.log.call_count == target_n_trials
 
 
+@pytest.mark.parametrize(
+    "metric,expected", [("value", ["x", "y", "value"]), ("foo", ["x", "y", "foo"])]
+)
 @mock.patch("optuna.integration.wandb.wandb")
-def test_values_registered_on_epoch(wandb: mock.Mock) -> None:
+def test_values_registered_on_epoch(wandb: mock.Mock, metric: str, expected: List[str]) -> None:
 
     wandb.log = mock.MagicMock()
 
-    wandbc = WeightsAndBiasesCallback()
+    wandbc = WeightsAndBiasesCallback(metric_name=metric)
     study = optuna.create_study()
     study.optimize(_objective_func, n_trials=1, callbacks=[wandbc])
 
     kall = wandb.log.call_args
-    assert list(kall[0][0].keys()) == ["x", "y", "value"]
+    assert list(kall[0][0].keys()) == expected
     assert kall[1] == {"step": 0}
+
+
+@pytest.mark.parametrize(
+    "metrics,expected",
+    [("value", ["x", "y", "value_0", "value_1"]), (["foo", "bar"], ["x", "y", "foo", "bar"])],
+)
+@mock.patch("optuna.integration.wandb.wandb")
+def test_multiobjective_values_registered_on_epoch(
+    wandb: mock.Mock, metrics: Union[str, List[str]], expected: List[str]
+) -> None:
+
+    wandb.log = mock.MagicMock()
+
+    wandbc = WeightsAndBiasesCallback(metric_name=metrics)
+    study = optuna.create_study(directions=["minimize", "maximize"])
+    study.optimize(_multiobjective_func, n_trials=1, callbacks=[wandbc])
+
+    kall = wandb.log.call_args
+    assert list(kall[0][0].keys()) == expected
+    assert kall[1] == {"step": 0}
+
+
+@pytest.mark.parametrize("metrics", [(["foo"]), (["foo", "bar", "baz"])])
```

Comment:
```suggestion
@pytest.mark.parametrize("metrics", [["foo"], ["foo", "bar", "baz"]])
```

Reply:
Thanks! Addressed in b3bc6f718ab0c879ec86d95cf40ef368f2b342a0